docs/release_notes/2024/August_2024/August_2024.md

```diff
@@ -9,31 +9,75 @@ tags:
   - august
 ---
 
+## 3.3.10.\* (August 30, 2024)
+
+- Prophecy Python libs version: 1.9.9
+- Prophecy Scala libs version: 8.0.31
+
+### Features {#Features3310}
+
+#### Spark Copilot Enhancements {#Spark3310}
+
+- **Source suggestion**: While working on your Pipeline, if there is an unconnected source, Data Copilot will now suggest other sources that you can add to it on the graph.
+
+#### SQL Copilot Enhancements {#SQL3310}
+
+- **SQL Subgraph**: We now support Subgraphs in your SQL models. Subgraphs allow you to wrap multiple different Gems under a single parent Gem. These logical groupings can simplify the Visual view of your model.
+
+- **Support for configurations in the Target Model**: The Target Model supports column selection and project-level configurations.
+
+- **Schedule button**: Just as you can schedule a Pipeline in Spark, you can now use the Schedule button to schedule a model in SQL. Opening the Schedule screen shows all existing Jobs. You can also create a new Job.
+
+- **Support dragging function parameters**: You can drag and drop function parameters in the Visual view of the expression builder.
+
+### UX Improvements {#UXImprovements3310}
+
+- **Support teams search**: You can use search on the teams select box during Fabric creation.
+
+- **SQL upload file button**: The SQL upload file button is now always present on the Environment tab.
+
+### Minor Improvements {#MinorImprovements3310}
+
+- **Fix with AI while saving Gems**: When saving a Gem, Data Copilot provides a one-click option to automatically resolve any diagnostic errors within the Gem.
+
+- **Auto describe Datasets**: We now automatically describe columns using interim data when you run your model and when you drop a Dataset from the Environment tab.
+
```

Comment:
UDFs and Macros in copilot expression suggestions: The copilot AI can now generate expressions using custom UDFs and Macros defined in Spark and SQL projects.

Reply:
Done.